### PR TITLE
Add Chat PushNotification

### PR DESCRIPTION
**Pull request by angellan-msft**
## Purpose

* Add the sample project "add-chat-push-notification" to guide Contoso user to enable push notification via chat SDK.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Test the code

Step 1: Create a Chat Thread with User A and User B. We will let User A receive push notifications from User B later. 
Step 2: Download this repo and finish all the prerequisites in QuickStart.
Step 3: Put User A’s <ACESS_TOEKN> and your <ACS_RESOURCE_ENDPOINT> into AppSettings.plist. 
Step 4: Set “Enable Bitcode” as “No” for two Pods targets – AzureCommunicationChat and Trouter. 
Step 5: Plug the IOS device into your mac, run the program and click “allow” when asked to authorize push notification on device. You should be able to see “Push Notification Started” printed on Console once the registration is successful. 
Step 6: Let User B sends a message to the thread created in step 1. You (User A) should be able to receive a push notification in your IOS device. 

## What to Check
Verify that the following are valid
N/A

### CODE REVIEW

**Review by tjprescott — CHANGES_REQUESTED** [review]

**Review by tjprescott — CHANGES_REQUESTED** [review]
Also, definitely add the Pods folder to gitignore which should scale down the file count significantly.

File: add-chat-push-notifications/SwiftPushTest/AppDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            
+            let semaphore = DispatchSemaphore(value: 0)
+            DispatchQueue.global(qos: .background).async { [weak self] in
+                guard let self = self else { return }
+                guard let chatClient = self.chatClient else { return }
+                
+                /* Please copy the below line of code if you want to implement an Advanced Version of PushNotification.
+                   As the SDK user, you are supposed to set the key handler on your end.
+                 */
+                chatClient.pushNotificationKeyHandler = appGroupPushNotificationKeyHandler
```
[the remark this answers is not in the document]
Comment (angellan-msft) [reply add-chat-push-notifications/SwiftPushTest/AppDelegate.swift]: Comments fixed.

File: add-chat-push-notifications/SwiftPushTest/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        // Please copy the below line if you want to implement an Advanced Version of PushNotification
+        let appGroupId = "group.microsoft.contoso"
+        // Please copy the below line if you want to implement an Advanced Version of PushNotification
+        let keyTag = "PNKey"
+        
+        do{
+            /*
+               Please copy the below line of code if you want to implement an Advanced Version of PushNotification.
+               As the SDK user, you are supposed to generate the push notification key handler on your end.
+               In this sample app, we use the default AppGroupPushNotificationKeyHandler class provided by chat SDK to generate a key handler.
```
[the remark this answers is not in the document]
Comment (angellan-msft) [reply add-chat-push-notifications/SwiftPushTest/AppDelegate.swift]: Comments updated to a more clear way.

File: add-chat-push-notifications/SwiftPushTest/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+            guard granted else { return }
+            self?.getNotificationSettings()
+            UNUserNotificationCenter.current().delegate = self
+
+        }
+    }
+
+    func getNotificationSettings() {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            print("Notification settings: \(settings)")
```
[the remark this answers is not in the document]
Comment (angellan-msft) [reply add-chat-push-notifications/SwiftPushTest/AppDelegate.swift]: Remove some useless print statements.

**Comment by angellan-msft** [comment]
> Also, definitely add the Pods folder to gitignore which should scale down the file count significantly.

Pods folder removed.

**Review by tjprescott — APPROVED** [review]

**Review by ddematheu2 — APPROVED** [review]

**Pull request merged**